slack integration part 1 (#6)
jakemulf/google-analytics-twitter-ads-deephaven: README.md

It may take a few days for you to get approved.

### Slack

> In order to collect data from Slack's API, you need to setup a Slack application. To start you can go to https://api.slack.com/apps/ and click the `Create New App` button. Give your app a name and assign it to your workspace.

Reviewer: ```suggestion
To collect data from Slack's API, you need to set up a Slack application. To start, you can go to https://api.slack.com/apps/ and click the **Create New App** button. Give your app a name and assign it to your workspace.
```

> On the next page, click the `Permissions` section underneath `Add features and functionality`. Scroll down to the `Scopes` section and click `Add an OAuth Scope`. This should open up a downdown menu.

Reviewer: my preference is not to use backticks on button names or menu items - bold or plain

Reviewer: ```suggestion
On the next page, click the `Permissions` section underneath `Add features and functionality`. Scroll down to the `Scopes` section and click `Add an OAuth Scope`. This should open up a dropdown menu.
```

Author: Cool sounds good, I'll update all of them

Add the following OAuth scopes: `channels:history`, `channels:read`, `groups:history`, `groups:read`, `im:history`, `im:read`, `mpim:history`, `mpim:read`

> Now scroll up to the `OAuth Tokens for Your Workspace` section and click the `Install to Workspace` button. On the next page press `Allow`.

Reviewer: ```suggestion
Now, scroll up to the `OAuth Tokens for Your Workspace` section and click the `Install to Workspace` button. On the next page, press `Allow`.
```

> If this was done right, you should now see your app in the `Apps` section of your Slack workspace. Invite the app to any channels you want to track metrics in `/invite @<AppName>`.

Reviewer: ```suggestion
If this was done correctly, you should see your app in your Slack workspace's `Apps` section. Invite the app to any channels you want to track metrics in: `/invite @<AppName>`.
```

Back in the `OAuth Tokens for Your Workspace` section, there should be a `Bot User OAuth Token` field. This is the value used in the `SLACK_API_TOKEN` environmental variable for this project.

> Lastly you need your channel IDs. These are simply found in the channel information in your workspace. You can optionally save one of these in the `SLACK_CHANNEL` environmental variable for this project.

Reviewer: ```suggestion
Lastly, you need your channel IDs. These are simply found in the channel information in your workspace. You can optionally save one of these in the `SLACK_CHANNEL` environmental variable for this project.
```

twitter_table = twitter_ads_main(start_date, end_date, date_increment)
```

This example collects data from Slack.

```
slack = get_channel_messages(SLACK_CHANNEL)
```

### Parquet reading and writing

> There are two helper methods in `./app.d/parquet_writer.py` that can be used to read and write parquet files, `write_tables` and `read_tables`. `write_tables` expects to receive a list of tables.

Reviewer: ```suggestion
There are two helper methods in `./app.d/parquet_writer.py` that can be used to read and write Parquet files, `write_tables` and `read_tables`. `write_tables` expects to receive a list of tables.
```
